### PR TITLE
adjustments to sudo call and log test

### LINKED ISSUE #97: Use entry in sudoers config instead of `--preserve-env` when starting backend process
Not all supported distros have a new enough version of `sudo` to support the `--preserve-env=` flag. We should rather use the more standard approach of preserving the environment through a line in the suoders config.

### PR DESCRIPTION
**Pull request by veggiesaurus**
Closes #97

@confluence do you think we need to warn users about the additional line in the sudoers file that is now needed?

### CODE REVIEW

**Comment by confluence** [comment]
I think that there should be some kind of warning. This may be a good time to add a changelog, so that it can be included in the package metadata. We should probably also add the changelog to the documentation.

Is there any way for an npm package to print messages for the user upon installation?

**Comment by confluence** [comment]
I wonder if a safer option would be 1) to give the user explicit control over the `sudo` arguments as well as the sudoers config, and 2) leave the previous setup as the default. This would make upgrades safe, and also makes sense because we're compensating for an *old* version of `sudo` rather than changing to accommodate a *new* version.

If we did that, we could just add a note to the extended configuration explaining what to do with an older version of `sudo`.

We could also automatically detect the `sudo` version and pick the correct arguments (instead of exposing this to the user).

**Comment by veggiesaurus** [comment]
> I wonder if a safer option would be 1) to give the user explicit control over the `sudo` arguments as well as the sudoers config, and 2) leave the previous setup as the default. This would make upgrades safe, and also makes sense because we're compensating for an _old_ version of `sudo` rather than changing to accommodate a _new_ version.

That's a good point. I've actually just tried bionic and centos7/8 docker images, and they all seem to have a new enough `sudo` version, so I suspect this is related to RHEL7 in particular.

**Comment by veggiesaurus** [comment]
@confluence I have now added the config option you suggested, please re-evaluate

File: config/example_sudoers_conf.stub [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # customise this file to fit your environment using visudo /etc/sudoers.d/carta_controller
 
+# ensure the auth token environmental variable is persisted across the sudo call
+Defaults env_keep += "CARTA_AUTH_TOKEN"
+
```
Comment (confluence) [inline config/example_sudoers_conf.stub]: This should no longer be in the example; it should be moved to the note.

**Review by confluence — CHANGES_REQUESTED** [review]

**Review by confluence — APPROVED** [review]

**Pull request merged**